pyresample/utils/cf.py

```diff
@@ -0,0 +1,497 @@
+    xy = dict()
+    if y is None and x is None:
+        for axis in ('x', 'y'):
+            xy[axis] = _guess_cf_axis_varname(nc_handle, variable, axis, type_of_grid_mapping)
+            if xy[axis] is None:
+                raise ValueError("Could not guess the name of the {} axis of the {}".format(
+                    axis, grid_mapping_variable))
+    else:
+        # y= and x= are provided by the caller. Check they are valid CF coordinate variables
+        #   The order is always (y,x)
+        xy['y'] = y
+        xy['x'] = x
+        for axis in ('x', 'y'):
+            if not _is_valid_coordinate_variable(nc_handle, xy[axis], axis, type_of_grid_mapping):
+                raise ValueError(
+                    "Variable x='{}' is not a valid CF coordinate variable for the {} axis".format(xy[axis], axis))
+
+    # we now have the names for the x= and y= coordinate variables: load the info of each axis separately
+    axis_info = dict()
+    for axis in ('x', 'y'):
+        axis_info[axis] = _load_cf_axis_info(nc_handle, xy[axis],)
+
+    # there are few cases where the x/y values loaded from the CF files cannot be
+    #   used directly in pyresample. We need a conversion:
+    for axis in ('x', 'y'):
+        axis_info[axis] = _convert_XY_CF_to_Proj(crs, axis_info[axis])
+
+    # transfer information on the axis to the cf_info dict()
+    for axis in ('x', 'y'):
+        cf_info[axis] = dict()
+        cf_info[axis]['varname'] = xy[axis]
+        for k in axis_info[axis].keys():
+            cf_info[axis][k] = axis_info[axis][k]
+
+    # sanity check: we cannot have different units for x and y
+    unit = axis_info['x']['unit']
+    if axis_info['x']['unit'] != axis_info['y']['unit']:
+        raise ValueError("Cannot have different units for 'x' ({}) and 'y' ({}) axis.".format(
+            axis_info['x']['unit'], axis_info['y']['unit']))
+
+    # prepare the AreaDefinition object
+    # =================================
+
+    # create shape
+    shape = (axis_info['y']['nb'], axis_info['x']['nb'])
+
+    # get area extent from the x and y info
+    extent = _get_area_extent_from_cf_axis(axis_info['x'], axis_info['y'])
+
+    # transform the crs objecto a proj_dict (might not be needed in future versions of pyresample)
+    proj_dict = crs.to_dict()
+
+    # finally prepare the AreaDefinition object
+    area_def = geometry.AreaDefinition.from_extent('from_cf', proj_dict, shape, extent,
+                                                   units=unit, )
+
+    return area_def, cf_info
+
+
+def _load_cf_area_severalVariables(nc_handle, ):
+    """ load several AreaDefitions from a netCDF/CF file """
+
+    def _indices_unique_AreaDefs(adefs):
+        """ use a classic dict-based strategy to
+            get the indices of unique AreaDefinitions in a list """
+
+        uniqs = dict()
+        for i, adef in enumerate(adefs):
+            if not adef in uniqs:  # this uses AreaDefinition.__eq__()
```

Comment:
```suggestion
            if not (adef in uniqs):  # this uses AreaDefinition.__eq__()
```
For clarity

Reply:
thanks. pre-commit told me to change it to `adef not in uniqs` and I did so now.